### DISCUSSION TITLE
Add hover intent option to `useHover` hook

### DISCUSSION THREAD

**Discussion by alexcarpenter** [adobe/react-spectrum#4438] [Ideas]
It would be nice to pass an optional argument to `useHover` for hover intent.

Using `useHover` within a navigation menu, and want to make sure folks are intending to interact with the element before triggering the hover immediately.

Reference: https://briancherne.github.io/jquery-hoverIntent/

**Comment by snowystinger** [comment]
We don't really have anything that uses hover as the primary interaction, as that has a lot of accessibility pitfalls and problems, especially for touch users. I don't think we'd want to do anything to encourage hover as a primary interaction. Can you expand more on your use case?
Specifically, if you're using our higher level hooks, how is hover coming into play?

**Reply by mhfaust** [reply]
The main reason I want a hover-intent is not for any immediate change to the element, but rather for pre-fetching resources that come in to play when the user clicks or otherwise interacts with the element beyond just hovering. For example, I have a list of project-names, and when the user clicks one of them, I need to fetch details about that project and present them in a project-details pane.  If I don't prefetch, then every single project-details peek requires a noticeable wait for the user. If I prefetch the project details on just a basic hover event, this results in far, far too many fetch calls as the user rapidly moves their cursor over the list of projects on route to clicking the one they intend to inspect, potentially overloading the server.  If I can detect hover intent, when the user slows down over a particular project-name, I can pre-fetch the project details in the background earlier than the click, and significantly reduce the time to load the details. This enhances the user experience greatly in certain contexts where the user needs to do quick-peeks at rich details before deciding on taking actions at lower levels in the resource hierarchy.

**Reply by snowystinger** [reply]
That makes sense. I imagine this would also be wanted for focusing links as well? Or does every focus result in pre-fetching? Maybe it'd make more sense to expose this on components which result in pre-loading (links) and joins together various behaviours into one API instead of needing people to watch for hover and for focus for example?

**Comment by alexcarpenter** [comment]
Using within a navbar. When the user is moving into the document with their mouse, i'd like to make sure they are intending to interact with the menu vs their mouse unintentionally triggering the isHovered boolean.

https://codesandbox.io/s/relaxed-tereshkova-l55dbi?file=/src/App.js

**Reply by snowystinger** [reply]
Sorry, what I mean is, is this cosmetic? or is this a primary form of interaction?

If it's cosmetic, then I would recommend handling it yourself. If it's a primary form of interaction, then we'd really need to review that before putting it into our hook for the reasons I mentioned before.

You could do it yourself with something along these lines https://codesandbox.io/s/boring-lovelace-dbulk9?file=/src/App.js

**Reply by alexcarpenter** [reply]
When thinking about the usage, I wouldn't see using useHover without the hover intent 🤔.

But I really appreciate your taking the time to show how to achieve this one our side of things 🙇🏼
